src/extensions/emulator/specHelper.ts

```diff
@@ -124,16 +105,5 @@ export function getNodeVersion(resources: Resource[]): string {
       )}. \n Only Node runtimes are supported.`
     );
   }
-  if (_.includes(versions, "nodejs10")) {
-    return "10";
-  }
-  if (_.includes(versions, "nodejs6")) {
-    EmulatorLogger.forEmulator(Emulators.FUNCTIONS).logLabeled(
-      "WARN",
-      "extensions",
-      "Node 6 is deprecated. We recommend upgrading to a newer version."
-    );
-    return "6";
-  }
-  return "8";
+  return Math.max(...versions);
```

Comment:
Hmmm - I like the idea of adding enforcement on what node versions extensions can be published with, but I think we should do it at publish time, not in the emulator. In this case, the user doesn't control what version of node their extensions target, so I think it better to err on the side of just using the version the extension asks for.

I would be in favor of adding a backend check during publishExtensionVersion that errors out if someone tries to publish with an odd Node version (or one that is not yet supported by GCF)

Reply:
agree with adding validations at publish time. I guess I'm just confused why we need the `Math.max(...versions);` So today do we allow functions with multiple node versions running under one extension? And can multiple emulated extensions use different node versions?

Reply:
You can technically set different node versions for each function (since the underlying resources are totally separate) - however, as a bit of a hack/simplification, we decided not to support that in the emulator because its a bad pattern and an edge case. To handle the case where a single extensions asks for multiple versions of node, we just default to the highest version they ask for. 

We _do_ support emulating different extensions in different versions of node. I think this is probably a hard requirement, too - we want users to be able to emulate any combo of extensions, and some won't work correctly (or at all) on the wrong version of node.

I'd also chose to do Math.max(versions) instead of just capping at 14 so that this doesn't become an annoyance to maintain. If we cap at 14, we need to remember to come back here when 16, 18, 20, etc are released.